db/db_iter.cc

```diff
@@ -961,6 +960,7 @@ void DBIter::Seek(const Slice& target) {
     }
     direction_ = kForward;
     ClearSavedValue();
+    saved_key_.Reserve(saved_key_.Size() - 8);
```

Comment:
Can we add a simple comment saying that this line purpose is to convert the internal key to a user key because FindNextUserEntry needs a UserKey to be stored in saved_key_